### ISSUE TITLE
GPLv3+?

### ISSUE THREAD

**Issue by caryoscelus** [zeronet-conservancy/zeronet-conservancy#113]
the [legally dubious vote](https://github.com/HelloZeroNet/ZeroNet/issues/2273) on old ZeroNet codebase triggered switching to GPLv3-only and for now we're keeping it for the whole product (new contributions are GPLv3+) before we figure out what happened here . it seems that there was only one contributor , @TheNain38 , who was in favour of GPLv3-only . however they also accepted Lax license , so even if we find that their contributions are still part of currently used code , we can (as far as we consider that form of will-expression valid anyway) license their work under a lax license while the whole will be GPLv3+

more concerning , however , is the fact that some contributors never seemed to have expressed their opinion on license change

**Comment by zeronet-legal** [comment]
@caryoscelus

> ZeroNet project has recently been informed of some license incompatibilities.

**I was who initiated the license change**. This voting is started and falsified by @imachug. Keep that in mind. However, since the overwhelming majority stated that GPLv3 license is desired, the voting is still valid even with the falsification for the reason stated below.

The original license GPLv2 (that become null and void due to license violations) doesn't stated neither the **only** nor the **or any later** version, so the license was not only null and void, but also unclear.  

According to Richard Stallman; "**As a consequence, code released under GPL version 2 only can't be merged with code released under GPL version 3 only.**" - https://www.gnu.org/licenses/identify-licenses-clearly.html

What this mean? Because ZeroNet did not stated neither the **only** nor the **or any later** version in the previous license, not making such a statement result in a GPLv2-only license. This means that, **re-licensing ZeroNet from GPLv2-only to GPLv3-only is not possible**.

As a result, **ZeroNet is licensed under GPLv3-or-later**.

**Comment by caryoscelus** [comment]
@zeronet-legal **please refrain from personal attacks here**

now , the majority of vote is not enough to relicense contributions from those who didn't sign up to it . most people signed up for relicensing to either GPLv3+ or lax (which would allow to include their code under lax license of their choice , while the whole is licensed under GPLv3 and any other version of it that would be compatible with such lax license — most like any future version of GPL) , but the outcome was decided and recorded in repository to be GPLv3 (despite the fact that the only GPLv3-only voting contributor also accepted lax) . that means only that contributions from that point on were licensed under that license and there is little we can do about relicensing them (those by nofish anyway) besides gradually rewriting them

the whole vote result (however you read it) cannot be applied to previous contributions beside from people who explicitly expressed approval . so the license status of 0net is not clear at this point

that said , `zeronet-conservancy` will be gradually moving away from the old codebase and will get rid from all license-dubious code (if there's any left) . new contributions are already GPLv3+

**Comment by caryoscelus** [comment]
here are the commits by the person who refused GPLv3+ , but accepted GPLv3-only or Lax :

https://github.com/zeronet-conservancy/zeronet-conservancy/commits?author=TheNain38

feel free to reach out to them and ask what license do they want and we'll put mentions of desired Lax license in the respective files that still have remnants of those commits

**Comment by emdee-net** [comment]
@zeronet-legal Please clarify for me: "re-licensing ZeroNet from GPLv2-only to GPLv3-only is not possible" conflicts with 
"As a result, ZeroNet is licensed under GPLv3-or-later" in my mind...

**Comment by zeronet-legal** [comment]
@emdee-net it is well clarified, but let me try to explain it again.

GPLv3-or-later **is not** GPLv3-only! 
> According to Richard Stallman; "As a consequence, code released under GPL version 2 only can't be merged with code released under GPL version 3 only." - https://www.gnu.org/licenses/identify-licenses-clearly.html

@caryoscelus your comments are completely nonsense. It is not possible to license zeronet under GPLv3-only for the reasons mentioned above. Please stop making statements without having knowledge about important issues, such as licensing. 

The "voting" about the license change was called as soon I have reported that the GPLv2-only license is **null and void**. However, before the voting my copy of ZeroNet already used GPLv3-or-later license as I couldn't change the license. Even if ZeroNet would change that time the license to say MIT, I can't be forced at all to make the change nor any other two thousand repositories who by then forked ZeroNet.

Because wasn't able update from a GPLv2-only license to GPLv3-only, I was legally forced by the GPL terms to upgrade the license to GPLv3-or-later. 

In other words, **you must use GPLv3-or-later license on your fork or yourself can not distribute ZeroNet!** Everyone who distributes ZeroNet under a GPLv3-only license is in violation of the terms of the said license. Keep that in mind @caryoscelus before you continue to distribute your copy.

As someone who also hold copyright to zeronet, I herein notify you @caryoscelus that this repository is in violation of the GPLv3-or-later license, your license will not terminate indefinitely as ZeroNet's previous GPLv2-only license did, but you are at this point required to replace the license in **30 days**. See section 11 of the GPLv3 license.

**Comment by caryoscelus** [comment]
@zeronet-legal  there's no such license as "GPLv3-or-later" . please educate yourself before trying to educate others . you cannot update from GPLv2-only to any license . the only thing you can do is to get a new license from those who have copyright over code

**Comment by zeronet-legal** [comment]
Listen to me very carefully. I going to send GitHub a report if you don't replace the license as I the copyright holder demand it. 

I'm getting very tired about spamming of your repository and claiming to be "official". You're a scammer who discovered that the ZeroNet project left abandoned and now you intentionally violate the GPL license. 

You better replace the license with GPLv3-or-later before you find yourself and your repository deleted from GitHub. 

You telling me there is no GPLv3-or-later license? :rofl: 
**https://www.gnu.org/licenses/identify-licenses-clearly.html**

**Comment by zeronet-legal** [comment]
@caryoscelus 
> you cannot update from GPLv2-only to any license

You can't be more stupid than what you're now. I doubt you even know what was the voting about in ZeroNet's license change. Kindly ask you to fuck yourself.

**Comment by emdee-net** [comment]
@zeronet-legal I don't know the history of any things like "voting" or how the voting "was falsified" and I'm curious about a simple explanation so I can understand what you are saying.

But more importantly, why do you say that the "GPLv2-only license is null and void" What did the licence do to invalidate itself?

And also could you explain why you are "someone who also hold copyright to ZeroNet" - I assume that the only copyright holder is the creator - Tomaz Kocsis (https://www.ted.com/talks/tamas_kocsis_the_case_for_a_decentralized_internet ).

**Comment by emdee-net** [comment]
@zeronet-legal If you are going to file a report with github you should also file one with nixpkgs = https://repology.org/project/zeronet-conservancy/versions fgaz@fgaz.me. The code is being packaged and distributed under the zeronet-conservancy name, with the current licence, and no CI testing https://github.com/zeronet-conservancy/zeronet-conservancy/issues/133
